configurator/src/main/java/org/picmg/configurator/StateSetTabController.java

```diff
@@ -119,6 +141,41 @@ void onSaveAsChangesAction(ActionEvent event) {
 	}
 	@Override
 	public void initialize(URL location, ResourceBundle resources) {
-		//TODO: fill out with needed initialize
+		// fire action events if focus is lost on our text fields - this allows the normal action handler
+		// to update and check values.
+		stateSetVendorNameTextField.focusedProperty().addListener(new ChangeListener<Boolean>() {
+			@Override public void changed(ObservableValue<? extends Boolean> arg0, Boolean oldValue, Boolean newValue) {
+				if (!newValue) { stateSetVendorNameTextField.fireEvent(new ActionEvent()); }}});
+
+		stateSetVendorIANATextField.focusedProperty().addListener(new ChangeListener<Boolean>() {
+			@Override public void changed(ObservableValue<? extends Boolean> arg0, Boolean oldValue, Boolean newValue) {
+				if (!newValue) { stateSetVendorNameTextField.fireEvent(new ActionEvent()); }}});
+
+		stateSetIdTextField.focusedProperty().addListener(new ChangeListener<Boolean>() {
+			@Override public void changed(ObservableValue<? extends Boolean> arg0, Boolean oldValue, Boolean newValue) {
+				if (!newValue) { stateSetIdTextField.fireEvent(new ActionEvent()); }}});
+
+
+		// bind images to their input constraints
+		vendorNameImage.visibleProperty().bind(Bindings.createBooleanBinding(() ->
+						stateSetVendorNameTextField.textProperty().getValueSafe().isBlank(),
+				stateSetVendorNameTextField.textProperty()));
+
+		vendorIANAImage.visibleProperty().bind(Bindings.createBooleanBinding(() ->
+						stateSetVendorIANATextField.textProperty().getValueSafe().isBlank(),
+				stateSetVendorIANATextField.textProperty()));
+
+		stateSetIDImage.visibleProperty().bind(Bindings.createBooleanBinding(() ->
```

Comment:
I think creating bindings for a component and then manually setting its state image is not allowed in javafx and will throw exceptions. These are a good start but not quite finished (e.g. if they are integer checks like the action conditions have). I suggest we go with either binding of setting in action

Reply:
Yes. Binding OR works. No more exceptions. Integer validation also working